### PR TITLE
Migrate Statistics Plots to Bootstrap on status monitor page

### PR DESCRIPTION
**Pull request by saraycp**
Move all the JavaScript code from the view to an assets file.
Instead of passing values to JS as parameters, it uses data attributes now.

**Large screen:**

Build Status Monitor   Open Build Service

**Small screen:**

Build Status Monitor   Open Build Service(1)

### CODE REVIEW

File: src/api/.jshintignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ app/assets/javascripts/webui/application/bento
 app/assets/javascripts/webui/application/cm2
 app/assets/javascripts/webui2/cm2
 app/assets/javascripts/webui2/*.min.js
+app/assets/javascripts/webui2/monitor.js
```
Comment (coolo) [inline src/api/.jshintignore]: why?

File: src/api/app/assets/stylesheets/webui2/monitor.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#building,
+#events,
+#jobs {
```
Comment (coolo) [inline src/api/app/assets/stylesheets/webui2/monitor.scss]: as this moved not into the global namespace of css, I would recommend to pick a little longer ids (or scope them)

File: src/api/app/assets/javascripts/webui2/monitor.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+           { data: data.waiting, label: "Ready to build", color: 5},
+           {  data: data.blocked, label: "Blocked build job", color: 6 } ],
+      {
+          series: {
+              stack: true,
+              lines: { show: true, steps: false, fill: true },
+          },
+          xaxis: { mode: 'time' },
+          yaxis: { max: data.jobs_max, position: "left", labelWidth: 25 },
+    legend: { noColumns: 3, position: "ne", container: "#legend-jobs" }
```
Comment (coolo) [inline src/api/app/assets/javascripts/webui2/monitor.js]: the indenting looks wrong

File: src/api/app/views/webui2/webui/monitor/_events.html.haml [hunk 4]
```diff
@@ -1 +1,33 @@
...
+  .col-md-12.col-lg-6.py-4.text-center
+    #building
+    %p.my-2 Workers
+    .d-flex.justify-content-center.flot_legend#legend-building
+    .plotspinner= image_tag('ajax-loader.gif')
+  .col-md-12.col-lg-6.py-4.text-center
+    #events
+    %p.my-2 Repositories to recalculate
+    .d-flex.justify-content-center.flot_legend#legend-events
+    .plotspinner= image_tag('ajax-loader.gif')
```
Comment (coolo) [inline src/api/app/views/webui2/webui/monitor/_events.html.haml]: better use bootstrap spinners - ajax-loader.gif is kind of bento only

**Comment by saraycp** [comment]
[Monitor Page: Statistical Plots](https://trello.com/c/CYgMFN8n/1153-monitor-page-statistical-plots)

File: src/api/app/views/webui2/webui/monitor/_events.html.haml (hunk 4 above)
Comment (dmarcoux) [reply src/api/app/views/webui2/webui/monitor/_events.html.haml]: We have an example for a spinner in the patterns library: https://obs-patterns.netlify.com/atoms/icons.html

File: src/api/app/assets/javascripts/webui2/monitor.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+function plotValues(data) {
+  eventsPlot(data);
+  buildingPlot(data);
+  jobPlot(data);
+}
+
+function eventsPlot(data) {
+  $.plot($("#events"), [ { data: data.squeue_high, label: "High", color: "red" },
```
[the remark this answers is not in the document]
Comment (hennevogel) [reply src/api/app/assets/javascripts/webui2/monitor.js]: This is just moved from bento, right @saraycp? Please let's not get carried away with fixing things before we understand what people want to do with this.

File: src/api/.jshintignore (hunk 1 above)
Comment (saraycp) [reply src/api/.jshintignore]: I just wanted to move the JS code from the view to its own assets file. I addressed some of the lint errors, however there are some more to fix and our goal is only migrating the view and not rewriting the code.

File: src/api/app/assets/stylesheets/webui2/monitor.scss (hunk 2 above)
Comment (saraycp) [reply src/api/app/assets/stylesheets/webui2/monitor.scss]: @coolo better now?

File: src/api/app/assets/javascripts/webui2/monitor.js (hunk 3 above)
Comment (saraycp) [reply src/api/app/assets/javascripts/webui2/monitor.js]: fixed

File: src/api/app/views/webui2/webui/monitor/_events.html.haml (hunk 4 above)
Comment (saraycp) [reply src/api/app/views/webui2/webui/monitor/_events.html.haml]: Thanks both, already changed. :eyes:

File: src/api/app/assets/javascripts/webui2/monitor.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (saraycp) [reply src/api/app/assets/javascripts/webui2/monitor.js]: Exactly @hennevogel. @dmarcoux I just wanted to move the JS code from the view to its own asstes file. Meanwhile I improved a few lines but it's not our goal right now. So I'll keep it like it is if you don't mind.

**Comment by saraycp** [comment]
> I found two issues.
> 
>     The plots are empty whenever I changes the architecture filter. The architecture is passed as the range in the query string.
>     When changing the timeframe filter, the architecture is always the same (even if I choose something else than x86_64, it's always x86_64)


@dmarcoux you are right. I'm having a look...

File: src/api/.jshintignore (hunk 1 above)
Comment (coolo) [reply src/api/.jshintignore]: then file an issue with the rest and take it as a comment here? Just adding it to the ignore file sounds dangerous

File: src/api/app/assets/javascripts/webui2/monitor.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (dmarcoux) [reply src/api/app/assets/javascripts/webui2/monitor.js]: No worries. Marking this as resolved.

File: src/api/.jshintignore (hunk 1 above)
Comment (hennevogel) [reply src/api/.jshintignore]: It was ignored before, it is ignored now. Doesn't sound very dangerous to me :)

Comment (coolo) [reply src/api/.jshintignore]: Well, all of webui was ignored - not so for webui2

**Pull request merged**